adapters/richaudience/richaudience.go

```diff
@@ -206,14 +234,12 @@ func parseImpExt(imp *openrtb2.Imp) (*openrtb_ext.ExtImpRichaudience, error) {
 	return &richaudienceExt, nil
 }
 
-func getMediaType(impId string, imps []openrtb2.Imp) openrtb_ext.BidType {
-	for _, imp := range imps {
-		if imp.ID == impId {
-			if imp.Video != nil {
-				return openrtb_ext.BidTypeVideo
-			}
-			return openrtb_ext.BidTypeBanner
+func getMediaType(impId string, imp openrtb2.Imp) openrtb_ext.BidType {
+	if imp.ID == impId {
+		if imp.Video != nil {
+			return openrtb_ext.BidTypeVideo
 		}
+		return openrtb_ext.BidTypeBanner
```

Comment:
This is an anti pattern. Usually, BidType should be set in the response body of the bidder. `response.seatbid[i].bid.ext.prebid.type` is where you should set the BidType in your bidder code and use it to [set TypedBid.](https://github.com/prebid/prebid-server/pull/2669/files#diff-f95090b50cc7f4380eb10531db0822ff98ed8c6b17b99175d224162f1c830267R173)

Reply:
Do you plan to resolve this @richaudience ?